### ISSUE TITLE
Client Side Tracking For Tasks

### ISSUE THREAD

**Issue by LennonR** [code-corps/code-corps-ember#1333]
# Problem

We don't currently have mixpanel tracking for understanding task creation and management.

## Subtasks

1. Viewed project tasks tab (properties: project_name, project_id)
2. Clicked create new task button (properties: project_name, project_id)
3. Created new task (properties: project_name, project_id)
4. Viewed project task (properties: project_name, project_id, task_id)
5. Added skill to task (properties: project_name, project_id)
6. Change task status (properties: project_name, project_id, task_id, moved_from_column, moved_to_column)
7. Assigned a user to a task (properties: project_name, project_id, assignee_user_name, task_id)
8. Edited task (if tasks are editable), (properties: project_name, project_id, task_id)

**Comment by snewcomer** [comment]
Is this something that the backend would record?

**Comment by LennonR** [comment]
I would recommend the frontend doing this if at all possible, via the mixpanel client library.

**Comment by JoshSmith** [comment]
I think it makes sense to list it here first and figure out what we should track on the backend. In some of these cases it will be a little of both.

For example, I think 1 and 2 are best done in the front-end, but 3-8 best done on backend. But there are variants here even still. For example:

> 8. Edited task

Could be tracked by tracking that the person clicked to get the editing UI, and _also track_ the actual `update` action on the server.

**Comment by JoshSmith** [comment]
We have a wrapper library we use in Ember that works with Segment, and then on the backend we wrote some custom tracking and some custom testing of the tracking for integration tests, too.

**Comment by LennonR** [comment]
@JoshSmith Sounds good.  Just make sure you're handling the user identification properly as that can be tricky in split client side/server stuff.  Also, the client side lib automatically sends a bunch of info like city, state, device, etc.  The server side doesn't.  That's why I prefer the client side lib when possible.

**Comment by JoshSmith** [comment]
Yeah, the only problem with the client-side tracking is that something like a `422` wouldn't be caught here, whereas the pipeline on the server only proceeds with a `200`. But yeah, where it makes sense, dive into client-side.
